src/EditorFeatures/Core/Extensibility/NavigationBar/NavigationBarItem.cs

```diff
@@ -30,21 +30,13 @@ internal abstract class NavigationBarItem : IEquatable<NavigationBarItem>
         /// <remarks>This can be empty for items whose location is in another document.</remarks>
         public ImmutableArray<TextSpan> Spans { get; }
 
-        /// <summary>
-        /// The span in the owning document corresponding to where to navigate to if the user selects this item in the
-        /// drop down.
-        /// </summary>
-        /// <remarks>This can be <see langword="null"/> for items whose location is in another document.</remarks>
-        public TextSpan? NavigationSpan { get; }
```

Comment:
remove entirely.  TS and F# do not need it (it's redundant with 'Spans' above).  And C# uses an entirely different mechanism so it can navigate both to locations in the file and potentially to other files (for partial cases).